lib/request_via/client.rb

```diff
@@ -3,7 +3,12 @@
 module RequestVia
 
   class Client
-    attr_reader :address, :net_http
+    attr_accessor :address
+    attr_reader :net_http, :map
+
+    Map = -> (client, path) {
+      client.clone.tap { |c| c.address = BuildURL.(c.address, path) }
```

Comment:
Avoid the usage of a setter, building a new object.

Related to the previous comment.